### DISCUSSION TITLE
composite operation

### DISCUSSION THREAD

**Discussion by CWK1985** [openseadragon/openseadragon#2588] [Q&A]
I have to show only the differences between 2 images (red and green). I am using the composite operation "xor". However, I see that the edges of the matching shapes are also shown in a different color (in this case: avg RGB of the 2 images 128,128,0). I was expecting that the matched portion will be transparent.  Is there a way to hide the matched portion through some filters or any other way?
![20240930_135158](https://github.com/user-attachments/assets/219dfc7a-8910-489e-878d-3afa9b2e6983)

**Comment by iangilman** [comment]
Are you using OpenSeadragon for this? I'm curious what the overall goal is... 

Anyway, yes, that's how xor works. I assume you've played with the other operations to see if one works for you? https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/globalCompositeOperation

**Comment by CWK1985** [comment]
My goal is to just show the differences between 2 images and hide the matched portion so as not to confuse the user. The difference operator also doesn't hide the matched portion. So I wanted to know if there is a way I can hide the matched portion of the 2 images and only show the differences.

**Reply by iangilman** [reply]
Do you need to use OSD for this? It wouldn't be hard to do it with direct canvas manipulation, but adding OSD in makes it trickier.

**Comment by CWK1985** [comment]
I can load multiple tiled images with zoom and other functionalities available in OSD. The other composite operations also seem to be easy to use. So wanted to know if there is a way to handle this unmatched region issue in OSD.
Thanks for your support.

**Reply by iangilman** [reply]
Cool. It's an interesting scenario! I'm not sure what to suggest... This isn't something I've heard of people doing with OSD before.

If it wasn't in OSD I would probably suggest some sort of post-processing to clean up after the xor. I suppose you could do that with OSD as well... Get the pixels of the canvas after each frame, manipulate them somehow (e.g. threshold away the bits you don't want) and put the manipulated pixels back into the canvas.

**Comment by CWK1985** [comment]
Thanks. I will try the post-processing.
